bindgen/ir/ty.rs

```diff
@@ -1099,6 +1100,23 @@ impl Type {
                         // within the clang parsing.
                         TypeKind::Opaque
                     } else {
+                        // Check if this type definition is an alias to a pointer of a `const
+                        // struct` with the same name and add the `_ptr` suffix to it to avoid name
+                        // collisions.
+                        if !ctx.options().c_naming {
+                            if let Some(inner_name) = inner_spelling
+                                .strip_prefix("const struct ")
```

Comment:
This feels a bit hackish, relying on libclang's serialization... Can we use `pointee_type()` and cursor equality instead? If not I guess this is ok...

Reply:
I did a commit using `pointee_type` but I don't get the cursor equality part, wdym?